pkg/gardener/shoot/extender/provider.go

```diff
@@ -309,9 +311,27 @@ func setMachineImage(provider *gardener.Provider, defMachineImgName, defMachineI
 	}
 }
 
+func correctOneNodeWorkerPools(provider *gardener.Provider, existingWorkers []gardener.Worker) {
+	existingWorkersMap := make(map[string]gardener.Worker)
+	for _, existing := range existingWorkers {
+		existingWorkersMap[existing.Name] = existing
+	}
+
+	for i := range provider.Workers {
+		alignedWorker := &provider.Workers[i]
+		if len(alignedWorker.Zones) != 1 {
+			continue
+		}
+
+		if existing, found := existingWorkersMap[alignedWorker.Name]; found {
```

Comment:
We ensure here that the Gardener will receive a proper value which should fix #674. 

The Runtime CR's `.provider.additionalWorkers` section will be however at the end different than shoots in terms of the zones. Should we not at least log a warning in such a case? I understand that at the end fixing https://github.com/kyma-project/kyma-environment-broker/issues/1766 should prevent this situation from re-occuring, but if there will be regression - we will not know about it.